### DISCUSSION TITLE
List all references for the "VBAproject"? (Stored in the "dir stream")

### DISCUSSION THREAD

**Discussion by rickefors** [decalage2/oletools#752] [Ideas]
Hi

This is a great collection of tools!
Can you please add functionality to list all references for the "VBAproject"?

They are stored in the "dir stream"
https://docs.microsoft.com/en-us/openspecs/office_file_formats/ms-ovba/2e06f06a-90ca-465f-b3fc-3e17c64dc491

Keep up the good work!!

**Comment by decalage2** [comment]
This is a good idea, related to the following issues: #286, #386, #707, #719
And BTW I think that you can already see references in the logs if you set the log level to debug with the option `-l debug`, as you can see in this commit: https://github.com/decalage2/oletools/commit/866ab3393878df01c2e15cdfdd2b2371aa894808

**Reply by rickefors** [reply]
Thanks for the help.

In Windows Command prompt I redirected STDERR (2) to add with the STDOUT (1)      "2>&1"
Then I piped the output to the find command (with the case insensitive flag set) and then searched for "reference"

olevba -l debug MyDoc.grf 2>&1 | find /i "reference"

I had a few different "reference type" 0016, 000D, 0033, 002F, 000F
Are these reference types documented somewhere ?
